packages/react/src/hooks/useGetURL.ts

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+
+import { getUrl, GetUrlInput, GetUrlOutput } from 'aws-amplify/storage';
+
+import { isFunction } from '@aws-amplify/ui';
+import { useHasValueUpdated } from '@aws-amplify/ui-react-core';
+
+type UseGetUrlInput = GetUrlInput & {
+  onError?: (error: Error) => void;
+};
+
+interface UseGetUrlOutput {
+  isLoading: boolean;
+  url?: URL;
+  expiresAt?: Date;
+}
+
+// interface UseGetUrlOutput extends GetUrlOutput{
+//     isLoading: boolean;
+// }
+
+function getKeyOrPath(
+  input: UseGetUrlInput
+): string | (({ identityId }: { identityId?: string }) => string) {
+  if (input.key !== undefined) {
+    return input.key;
+  }
+  return input.path;
+}
+
+type UseGetUrl = (input: UseGetUrlInput) => UseGetUrlOutput;
+
+export const useGetURL: UseGetUrl = (
+  input: UseGetUrlInput
+): UseGetUrlOutput => {
+  const [result, setResult] = React.useState<GetUrlOutput>();
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const imgLocation = getKeyOrPath(input);
+  const hasImgUpdated = useHasValueUpdated(imgLocation);
+  const { onError, ...getUrlInput } = input;
```

Comment:
`getUrlInput` will have a new object reference on each call of this hook causing the `useEffect` on line 42 to run on every render. Anything we can do to prevent that?

Reply:
Yeah I think we can just define it inside of the `useEffect`